Program.fs

```diff
@@ -18,11 +18,14 @@ type LibraryWithNpmDeps = {
     NpmDependencies : NpmDependency list
 }
 
+
 [<RequireQualifiedAccess>]
 type ResolveAction =
-    | Install of package:string * library:string * version:string
-    | Uninstall of package:string * library: string * version:string
-    | UnableToResolve of package:string * library:string * range:string * error: string
+    | Install of library:string * package:string * version:string * range: string
+    | InstallDev of library:string * package:string * version:string * range: string
+    | Uninstall of library:string * package: string * version:string
+    | UninstallDev of library:string * package: string * version:string
+    | UnableToResolve of library:string * package:string * range:string * error: string
```

Comment:
I am not a huge fan of using DUs with a lot of arguments because it makes it hard to access a property.

If we were replacing it with something like:

```fs
type InstallInfos =
    { 
        Package : string
        Library : string
        Version : string 
        Range : string
        IsDev : bool
    }

type UninstalInfos =
    { 
        Package : string
        Library : string
        Version : string 
        IsDev : bool
    }

type UnableToResolveInfos =
    {
        Library : string
        Package : string
        Range : string
        Error : string
    }

type ResolveAction =
    | Install of InstallInfos
    | Uninstall of UninstalInfos
    | UnableToResolve of UnableToResolveInfos
```

It would resolve the code duplication

Reply:
Yeah I agree, the code needs a cleanup and using records will make it more readable / more compact. But since this is just a refactoring that will require quite some work, I will defer it to another issue, first I want to get this feature in :smile: (see #37)